pkg/expression/helper.go

```diff
@@ -89,8 +89,11 @@ func getTimeCurrentTimeStamp(ctx sessionctx.Context, tp byte, fsp int) (t types.
 // GetTimeValue gets the time value with type tp.
 func GetTimeValue(ctx sessionctx.Context, v interface{}, tp byte, fsp int, explicitTz *time.Location) (d types.Datum, err error) {
 	var value types.Time
+	tc := ctx.GetSessionVars().StmtCtx.TypeCtx()
+	if explicitTz != nil {
+		tc = tc.WithLocation(explicitTz)
```

Comment:
`tc.WithLocation(explicitTz)` will create a new context with old one not modified. So there is no data race any more